### PR TITLE
Upgrade node and deps

### PR DESCRIPTION
**Pull request by edizcelik**
### Description

- ~~Node version is upgraded to `14.9.0` as well~~. Node version is set to `12.18.3`.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -2,7 +2,7 @@ version: 2
 jobs:
   deploy_staging:
     docker:
-      - image: circleci/node:12.16.1
+      - image: circleci/node:14.9.0
```
Comment (mucahit) [inline .circleci/config.yml]: Do you see anything that can be risky to upgrade 14.x? 

I was thinking we'll stay on the 12.x version of the Node till 14.x becomes LTS.

**Review by mucahit — APPROVED** [review]

**Review by burakguneli — APPROVED** [review]

File: .circleci/config.yml (hunk 1 above)
Comment (edizcelik) [reply .circleci/config.yml]: I was thinking the same. I'm not sure really, should I downgrade ?

Comment (mucahit) [reply .circleci/config.yml]: Nodejs says;
>  Production applications should only use Active LTS or Maintenance LTS releases.


Let's stay on the 12.x then. 

14.x will be active LTS on 2020-10-27, we can do it next month

Comment (edizcelik) [reply .circleci/config.yml]: Cool I will make the changes

**Pull request merged**